src/ui/public/modals/confirm_modal.js

```diff
@@ -23,9 +27,15 @@ module.factory('confirmModal', function ($rootScope, $compile) {
   return function confirmModal(message, customOptions) {
     const defaultOptions = {
       onCancel: noop,
-      cancelButtonText: 'Cancel'
+      cancelButtonText: 'Cancel',
+      onClose: noop,
```

Comment:
> This also introduces a gotcha where a user can forget to add an onCancel handler even in situations where they DID add an onCancel,

I'm not sure what you mean by this.

View/edit mode needs this situation.  It has two choices and I also want to give the ability for the user to "x" out of the dialog and "do nothing".  I suppose I could not accept a custom onClose function and always make it a noop, which would be sufficient for my needs.

These are the two dialogs that I want to use this for:
Here 'x' will cancel out jumping into edit mode:
<img width="670" alt="screen shot 2017-01-17 at 4 29 50 pm" src="https://cloud.githubusercontent.com/assets/16563603/22315728/8a244988-e336-11e6-9dfc-e4cc94e0e761.png">

Here 'x' will cancel out leaving edit mode:
<img width="585" alt="screen shot 2017-01-17 at 4 30 55 pm" src="https://cloud.githubusercontent.com/assets/16563603/22315729/8a2571b4-e336-11e6-8115-e18d218fa5ee.png">